test/e2e/main.test.ts

```diff
@@ -343,5 +344,38 @@ describe("E2E", function () {
         postFundBalance
       );
     });
+
+    it("should have converted all vested tokens and send them to upkeep", async function () {
+      await paymentAdapter.file(
+        formatBytes32String("bufferMax"),
+        parseEther("100")
+      );
+
+      const tx = await keeperRegistryPerformUpkeep(
+        registry,
+        registrySigners,
+        upkeepId,
+        KeeperRegistryParams.F
+      );
+
+      const topUpEventABI = [
+        "event TopUp(uint256 bufferSize, uint256 daiBalance, uint256 daiSent)",
+      ];
+      const { daiSent } = parseEventFromABI(tx, topUpEventABI);
```

Comment:
Pleases check `getLinkSwapped` function from above.

Also refactor all event decodings in the file to use the new util function.